server/routes/addPatient/prisonerSelect.ts

```diff
@@ -33,18 +30,41 @@ export default class PrisonerSelectRoutes {
 
     const searchResults = await this.prisonerSearchService.search({ searchTerm, prisonIds: ['OUT'] }, user)
 
-    const missingPatients = searchResults.filter(prisoner => this.searchFilter.includePrisonerToAdd(prisoner))
+    const missingPatients = searchResults
+      .map(prisoner => {
+        return { ...prisoner, searchStatus: this.searchFilter.includePrisonerToAdd(prisoner) }
+      })
+      .filter(prisoner => prisoner.searchStatus !== SearchStatus.EXCLUDE)
+      .map(prisoner => {
+        return { ...prisoner, actionLink: this.addLink(prisoner, searchTerm) }
+      })
 
-    return this.renderView(req, res, { searchResults: missingPatients, searchTerm })
+    return this.renderView(req, res, { searchResults: missingPatients })
   }
 
   submit = async (req: Request, res: Response): Promise<void> => {
     const { searchTerm } = req.body
 
     const error = validateForm({ searchTerm })
 
-    if (error) return this.renderView(req, res, { error, searchTerm })
+    if (error) return this.renderView(req, res, { error })
 
     return res.redirect(`/add-restricted-patient/select-prisoner?${new URLSearchParams({ searchTerm })}`)
   }
+
+  private addLink(prisoner: PrisonerSearchSummary, searchTerm: string): string {
+    if (prisoner.searchStatus === SearchStatus.INCLUDE) {
+      return `<a href="/add-restricted-patient/select-hospital?prisonerNumber=${prisoner.prisonerNumber}&journeyStartUrl=/add-restricted-patient/select-prisoner?searchTerm=${searchTerm}" class="govuk-link" data-test="prisoner-add-restricted-patient-link"><span class="govuk-visually-hidden">${prisoner.displayName} - </span>Add to restricted patients</a>`
+    }
+    if (prisoner.searchStatus === SearchStatus.EXCLUDE_POST_CRD) {
+      return `<p><span class="govuk-visually-hidden">${prisoner.displayName} - </span>Ineligible (past CRD)<br><a href="/help?section=restricted-patients-should-be-removed" class="govuk-link" data-test="help-link" target="restricted_patients_help">View Help</a></p>`
+    }
+    if (prisoner.searchStatus === SearchStatus.EXCLUDE_POST_SED) {
+      return `<p><span class="govuk-visually-hidden">${prisoner.displayName} - </span>Ineligible (past SED)<br><a href="/help?section=restricted-patients-should-be-removed" class="govuk-link" data-test="help-link" target="restricted_patients_help">View Help</a></p>`
+    }
+    if (prisoner.searchStatus === SearchStatus.EXCLUDE_NOT_RELEASED_HOSPITAL) {
+      return `<p><span class="govuk-visually-hidden">${prisoner.displayName} - </span>Ineligible (not released to hospital)<br><a href="/help?section=not-released-to-hospital" class="govuk-link" data-test="help-link" target="restricted_patients_help">View Help</a></p>`
+    }
+    return ''
+  }
```

Comment:
A bit of a knee-jerk comment - but cant you just pass the data and render the html in the njk file?

Reply:
It's a fair point. I started out that way but found that the njk file was really hard to read and I lost the value of using enums for each case. I think that minimal logic is OK in a template but too much needs pushing up into code.